packages/atlas/src/components/_inputs/TextFieldWithDropdown/TextFieldWithDropdown.tsx

```diff
@@ -0,0 +1,106 @@
+import { useCombobox } from 'downshift'
+import React, { useEffect, useRef, useState } from 'react'
+
+import { ListItem, ListItemProps } from '@/components/ListItem'
+import { SvgActionLoader } from '@/components/_icons'
+
+import { ListWrapper, TextFieldWithDropdownWrapper } from './TextFieldWithDropdown.styles'
+
+import { TextField, TextFieldProps } from '../TextField'
+
+export type TextFieldWithDropdownProps<ItemType = unknown> = {
+  items?: (ListItemProps & ItemType)[]
+  loading?: boolean
+  onSelectedItemChange?: (item?: ListItemProps & ItemType) => void
+  onInputValueChange?: ((item?: string) => void) | ((item?: string) => Promise<void>)
+  resetOnSelect?: boolean
+  notFoundNode?: ListItemProps | null
+} & Omit<TextFieldProps, 'charactersCount'>
+
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-constraint
+export const TextFieldWithDropdown = <ItemType extends unknown>(props: TextFieldWithDropdownProps<ItemType>) => {
```

Comment:
Maybe we could use a simpler name for this component? `Combobox` seems to be actually quite popular, what do you think about this name?

Reply:
Agree, updated